api/client/trust.go

```diff
@@ -324,19 +345,38 @@ func (cli *DockerCli) trustedPull(repoInfo *registry.RepositoryInfo, ref registr
 		if err != nil {
 			return notaryError(repoInfo.FullName(), err)
 		}
+		repoHasReleasesRole, err := hasReleasesRole(notaryRepo)
+		if err != nil {
+			return notaryError(repoInfo.FullName(), err)
+		}
```

Comment:
Can you move this out of the `if`, since it's done in both the `if` and `else` cases?